providers/openfeature-flagd-provider/lib/openfeature/flagd/provider.rb

```diff
@@ -7,12 +7,12 @@
 require_relative "provider/version"
 
 module OpenFeature
```

Comment:
Should this PR be taken as an opportunity to fix the main namespace as well?

`openfeature` folder should result in a `Openfeature` constant, otherwise the folder could be renamed to `open_feature` (maybe makes more sense)?

Reply:
I think (similar to the Python and JS SDKs) we should break the Ruby convention and continue to use `OpenFeature` as the top-level module. We can handle the file naming discrepancy in the loading, but this matches the `openfeature-sdk` published artifact better.

Reply:
ok if everything is aligned on this rule, cross language, that makes sense not to break it